src/jit/compiler.h

```diff
@@ -604,13 +638,19 @@ class  LclVarDsc
     bool                lvIsMultiregStruct()
     {
 #if FEATURE_MULTIREG_ARGS_OR_RET
-#ifdef _TARGET_ARM64_
-        if ((TypeGet() == TYP_STRUCT) &&
-            (lvSize()  == 2 * TARGET_POINTER_SIZE))
+        if (TypeGet() == TYP_STRUCT)
         {
-            return true;
+            if (lvIsHfa() && (lvHfaSlots() > 1))
```

Comment:
I will put back this guard and change lvHfaSlots to assert that it is dealing with a HFA